docs/source/en/model_doc/clvp.md

```diff
@@ -0,0 +1,128 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# CLVP
+
+## Overview
+
+The CLVP (Contrastive Language-Voice Pretrained Transformer) model was proposed in [Better speech synthesis through scaling](https://arxiv.org/abs/2305.07243) by James Betker.
+
+The abstract from the paper is the following:
+
+*In recent years, the field of image generation has been revolutionized by the appli-
+cation of autoregressive transformers and DDPMs. These approaches model the
+process of image generation as a step-wise probabilistic processes and leverage
+large amounts of compute and data to learn the image distribution.
+This methodology of improving performance need not be confined to images. This
+paper describes a way to apply advances in the image generative domain to speech
+synthesis. The result is TorToise - an expressive, multi-voice text-to-speech sys-
+tem.*
+
+Tips:
+
+1. CLVP or Contrastive Language-Voice Pretrained Transformer is an integral part of the Tortoise TTS model.
+2. CLVP can be used to compare different generated speech candidates with the provided text, and the best speech tokens are forwarded to the diffusion model.
+3. The use of the [`CLVPModel.generate()`] method is strongly recommended for tortoise usage
+
+Brief Explanation:
+
+First the [`CLVPTokenizer`] tokenizes the `text` and [`CLVPFeatureExtractor`] extracts `log-melspectrogram` from the
+desired `audio`. Then both of the text tokens and audio representations are passed to the [`CLVPConditioningEncoder`] 
+which converts both of them into `conditioning_embeds` which preserves the text information as well as the properties 
+of the voice. [`CLVPAutoRegressiveLMHeadModel`] uses those embeds to generate multiple `speech candidates`. 
+Now the speech encoder converts each speech candidate into a vector representation and the text encoder converts the 
+text tokens into a vector with similar dimension. At the end we compare each speech vector with the text vector to see
+which speech vector is most similar to the text.
```

Comment:
There is something weird with the layout haha:
```suggestion
Firstly, the [`CLVPTokenizer`] generates the `text` and the [`CLVPFeatureExtractor`] extracts the `log-melspectrogram` from the desired `audio`. 
Next, the text tokens and audio representations are passed to the [`CLVPConditioningEncoder`], which converts them into `conditioning_embeds` that preserve the text information as well as the voice properties. The [`CLVPAutoRegressiveLMHeadModel`] uses these embeds to generate multiple `speech candidates`. 
The speech encoder converts each speech candidate into a vector representation, and the text encoder converts the text tokens into the same latent space. 
At the end, we compare each speech vector with the text vector to see which speech vector is most similar to the text vector. 
```

Reply:
> seems pretty similar to CLAP ? (Just asking for my information! )

Yes kind of same but instead of comparing `audio` vs `text`, here we compare `speech tokens` vs `text`. `speech tokens` have discrete values(like text tokens) and generated by the decoder model(`gpt2`). CLVP just compares the `text` with the generated `speech tokens`. 
This is quite important for `Tortoise-TTS` since the diffusion model uses those `speech tokens` to generate the final speech. So we want to filter out the best of the best `speech tokens` from all the generated tokens. And the filtration is done by CLVP.